### PR TITLE
JSDoc improvements

### PR DESCRIPTION
**Pull request by emmenko**
**do not merge yet**

### CODE REVIEW

**Comment by emmenko** [comment]
@gaearon I've had a look at the JSDocs and improved them a little (hopefully). Otherwise LGTM :+1:

File: docs/Reference/API.md [hunk 1]
```diff
@@ -48,9 +48,9 @@ There should only be a single store in your app.
 * `reducer: Reducer`: Required. A [reducer](./Glossary.md#reducer) function that returns the next state tree, given
 the current state tree and the action to handle.
 
-* `initialState: any`: The initial [state](./Glossary.md#state). You may optionally specify it
+* `initialState: Object`: The initial [state](./Glossary.md#state). You may optionally specify it
 to hydrate the state from the server in universal apps, or to restore a
-previously serialized user session.
+previously serialized user session. The object keys must match the reducer names managed by `combineReducers`.
```
Comment (gaearon) [inline docs/Reference/API.md]: This is a tricky one. Please don't *have to* use `combinedReducers`. They may be passing a single reducer function that operates on an Immutable tree. Maybe rephrasing it as “If you use `combineReducers` to produce the root reducer function, this must be an object with the same shape as `combineReducers` keys.” would help.

File: src/createStore.js [hunk 2]
```diff
@@ -125,6 +127,11 @@ export default function createStore(reducer, initialState) {
     dispatch({ type: ActionTypes.INIT });
   }
 
+  /**
```
Comment (gaearon) [inline src/createStore.js]: This is not really JSDoc, is it? It's describing a line of code, not a variable or export. Should it have `/**`?

File: docs/Reference/API.md (hunk 1 above)
Comment (emmenko) [inline docs/Reference/API.md]: Makes sense, will rephrase it :+1:

File: src/createStore.js (hunk 2 above)
Comment (emmenko) [inline src/createStore.js]: Hmm good question. No it's not meant to be JSDoc, just a comment to explain why we are dispatching. Should I use `//` line comments or should I just remove it? What do you suggest?

Comment (gaearon) [inline src/createStore.js]: I think `//` is fine in these cases.

**Comment by emmenko** [comment]
@gaearon ok, done the changes.

File: docs/Reference/API.md [hunk 3]
```diff
@@ -48,9 +48,9 @@ There should only be a single store in your app.
 * `reducer: Reducer`: Required. A [reducer](./Glossary.md#reducer) function that returns the next state tree, given
 the current state tree and the action to handle.
 
-* `initialState: any`: The initial [state](./Glossary.md#state). You may optionally specify it
+* `initialState: Object`: The initial [state](./Glossary.md#state). You may optionally specify it
```
Comment (gaearon) [inline docs/Reference/API.md]: It still doesn't have to be an object. `any` is correct.

File: src/createStore.js [hunk 4]
```diff
@@ -22,9 +22,11 @@ export var ActionTypes = {
  * @param {Function} reducer A function that returns the next state tree, given
  * the current state tree and the action to handle.
  *
- * @param {any} initialState The initial state. You may optionally specify it
+ * @param {Object} [initialState] The initial state. You may optionally specify it
```
Comment (gaearon) [inline src/createStore.js]: Same: `any` is correct.

File: src/createStore.js [hunk 5]
```diff
@@ -125,6 +128,10 @@ export default function createStore(reducer, initialState) {
     dispatch({ type: ActionTypes.INIT });
   }
 
+
+  // When a store is created, an "INIT" action is dispatched so that every
+  // reducer will return their initial state. This effectively populates
+  // the initial store tree.
```
Comment (gaearon) [inline src/createStore.js]: "store tree" -> "state tree".

File: src/createStore.js [hunk 6]
```diff
@@ -125,6 +128,10 @@ export default function createStore(reducer, initialState) {
     dispatch({ type: ActionTypes.INIT });
   }
 
+
+  // When a store is created, an "INIT" action is dispatched so that every
+  // reducer will return their initial state. This effectively populates
```
Comment (gaearon) [inline src/createStore.js]: "will return" -> "returns" (grammar :-)

File: src/utils/isPlainObject.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /**
- * @param {any} obj The object to inspect.
- * @returns {boolean} True if the argument appears to be a plain object.
+ * @param {*} obj The object to inspect.
```
Comment (gaearon) [inline src/utils/isPlainObject.js]: I don't mind changing `any` to `*` but then this needs to be done everywhere (including API reference doc).
I'm also happy with keeping it called `any` :-)

File: src/createStore.js (hunk 4 above)
Comment (emmenko) [inline src/createStore.js]: But shouldn't we expect an `Object` here?

File: src/utils/isPlainObject.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
...
+ * @returns {Boolean} True if the argument appears to be a plain object.
```
Comment (gaearon) [inline src/utils/isPlainObject.js]: Why change this? `boolean` is lowercase in Google Closure types reference (one of JSDoc type “providers”), it is also lowercase in Flow.

File: src/createStore.js (hunk 5 above)
Comment (emmenko) [inline src/createStore.js]: :+1:

File: src/createStore.js (hunk 6 above)
Comment (emmenko) [inline src/createStore.js]: :+1:

File: src/createStore.js (hunk 4 above)
Comment (gaearon) [inline src/createStore.js]: Nope. Only if you use `combineReducers`. Otherwise it can be anything, an array, an Immutable object, etc.

Comment (gaearon) [inline src/createStore.js]: Also it can be `undefined`. It's optional

File: src/utils/isPlainObject.js (hunk 7 above)
Comment (emmenko) [inline src/utils/isPlainObject.js]: Hmm ok, I just taken it from http://usejsdoc.org/tags-param.html

screen shot 2015-07-27 at 16 10 25

File: src/utils/isPlainObject.js (hunk 8 above)
Comment (emmenko) [inline src/utils/isPlainObject.js]: My bad, I thought it was always uppercase...will revert.

File: src/createStore.js (hunk 4 above)
Comment (emmenko) [inline src/createStore.js]: > it can be anything, an array, an Immutable object, etc.

Riiiight, did thought of that. Thanks! :+1:

**Comment by emmenko** [comment]
@gaearon I've squashed the commits. I think it should be ok now. 

Regarding `any` vs `*` I'm not sure, I guess we can keep `any` :+1:

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**